dot/sync/chain_processor.go

```diff
@@ -258,15 +261,20 @@ func (s *chainProcessor) handleBlock(block *types.Block, announceImportedBlock b
 		return err
 	}
 
-	rt.SetContextStorage(ts)
+	nextStorageState := ts.Snapshot()
+	rt.SetContextStorage(nextStorageState)
 
 	_, err = rt.ExecuteBlock(block)
 	if err != nil {
+		rt.SetContextStorage(nil)
 		return fmt.Errorf("failed to execute block %d: %w", block.Header.Number, err)
 	}
 
-	if err = s.blockImportHandler.HandleBlockImport(block, ts, announceImportedBlock); err != nil {
-		return err
+	rt.SetContextStorage(nil)
```

Comment:
could we not just defer this line before line 267

Reply:
We could, I did that more-explicit-less-deferred to highlight the storage is no longer needed right after `rt.ExecuteBlock(block)`